DSCResources/MSFT_xRobocopy/MSFT_xRobocopy.psm1

```diff
@@ -13,15 +13,53 @@ function Get-TargetResource
         $Destination,
 
         [System.String]
-        $LogOutput
-    )
+        $Files,
+
+        [System.UInt32]
+        $Retry,
+
+        [System.UInt32]
+        $Wait,
+
+        [System.Boolean]
+        $SubdirectoriesIncludingEmpty,
+
+        [System.Boolean]
+        $Restartable,
+
+        [System.Boolean]
+        $MultiThreaded,
+
+        [System.String]
+        $ExcludeFiles,
+
+        [System.String]
+        $ExcludeDirs,
+
+        [System.String]
+        $LogOutput,
+
+        [System.Boolean]
+        $AppendLog,
 
-    $LogOutputExists = Test-Path $PSBoundParameters.LogOutput
+        [System.String]
+        $AdditionalArgs
+    )
 
     $returnValue = @{
-        Source = if (test-path $Source){[System.String]$(ls $Source).psChildName};
-        Destination = if (test-path $Destination){[System.String]$(ls $Destination).psChildName};
-        LogOutput = if ($LogOutputExists -eq $true){[System.String]$PSBoundParameters.LogOutput}
+        Source = $Source
+        Destination = $Destination
+        Files = $Files
+        Retry = $Retry
+        Wait = $Wait
+        SubdirectoriesIncludingEmpty = $SubdirectoriesIncludingEmpty
+        Restartable = $Restartable
+        Multithreaded = $MultiThreaded
+        ExcludeFiles = $ExcludeFiles
+        ExcludeDirs = $ExcludeDirs
+        LogOutput = $LogOutput
+        AppendLog = $AppendLog
+        AdditionalArgs = $AdditionalArgs

```

Comment:
Get-TargetResource is supposed to return the actual state of the machine.
In current implementation it returns whatever user wants it to be (whatever was specified in the configuration), but without saying what is the real state. 

I'd suggest following pattern used by other resources, adding readonly property "Ensure" which would contain information (Boolean) if file described in Destination path exists on the given node or not.